tests/testthat/test-use-rcpp.R

```diff
@@ -18,6 +18,39 @@ test_that("use_rcpp() creates files/dirs, edits DESCRIPTION and .gitignore", {
   expect_true(all(c("*.o", "*.so", "*.dll") %in% ignores))
 })
 
+test_that("use_rcpp_armadillo() creates Makevars files and edits DESCRIPTION", {
+  skip_if_not_installed("RcppArmadillo")
```

Comment:
I am concerned about skipping these and not adding to DESCRIPTION. I don't particularly want these as heavy dependencies, but I do want the tests to run on travis. Maybe `use_dependency()` needs some infrastructure to facilitate testing?

Reply:
As the Travis-CI builds are run by compiling all packages, I wanted to avoid adding another compilation requirement. Having said this, there is precedence elsewhere in `usethis` for:

1. adding packages to suggest _without_ skipping tests c.f. `pkgdown`. 
2. using test skips when a package is unavailable, c.f. `use_logo()` when `magick` cannot be found.

For now, I've opted to go the first route and have added the packages to the `Suggests` field of the `DESCRIPTION`. I say this because I'm not sure a valid solution would be to modify `use_dependency()` as the requirements should be kept as script as possible.